src/service/extensions-impl.js

```diff
@@ -263,18 +259,23 @@ export class Extensions {
     els.forEach((el) =>
       el.setAttribute('i-amphtml-loaded-new-version', extensionId)
     );
-    const urlParts = parseExtensionUrl(els[0].src);
-    return this.preloadExtension(extensionId, urlParts?.extensionVersion);
+    return this.preloadExtension(extensionId, version);
   }
 
   /**
    * @param {!Window} win
    * @param {string} extensionId
    * @param {string=} version
+   * @param {boolean=} latest
    * @return {!Promise}
    */
-  importUnwrapped(win, extensionId, version = DEFAULT_VERSION) {
-    const scriptsInHead = getExtensionScripts(win, extensionId);
+  importUnwrapped(win, extensionId, version = DEFAULT_VERSION, latest = true) {
+    const scriptsInHead = getExtensionScripts(
+      win,
+      extensionId,
+      version,
+      latest
+    );
     let scriptElement = scriptsInHead.length > 0 ? scriptsInHead[0] : null;
```

Comment:
Oh, just noticed this bug. `scriptsInHead` can be an array of `[js, mjs]` scripts, and we don't check for which we actually care about based on the Runtime's ESM setting.

Reply:
But does it matter in this context? Or can the `.js`/`.mjs` scripts get out of sync?

Reply:
On an ESM supporting browser, the `.js` version will never load. And vice-versa. I think we have to install the listener on both, else we could get a never ending promise.

Reply:
Will be fixed separately.